src/pip/_internal/commands/cache.py

```diff
@@ -0,0 +1,152 @@
+from __future__ import absolute_import
+
+import logging
+import os
+import textwrap
+
+import pip._internal.utils.filesystem as filesystem
+from pip._internal.cli.base_command import Command
+from pip._internal.cli.status_codes import ERROR, SUCCESS
+from pip._internal.exceptions import CommandError, PipError
+from pip._internal.utils.typing import MYPY_CHECK_RUNNING
+
+if MYPY_CHECK_RUNNING:
+    from optparse import Values
+    from typing import Any, List
+
+
+logger = logging.getLogger(__name__)
+
+
+class CacheCommand(Command):
+    """
+        Inspect and manage pip's cache.
+
+        Subcommands:
+
+        info:
+            Show information about the cache.
+        list:
+            List filenames of packages stored in the cache.
+        remove:
+            Remove one or more package from the cache.
+        purge:
+            Remove all items from the cache.
+
+        <pattern> can be a glob expression or a package name.
+    """
+
+    usage = """
+        %prog info
+        %prog list [name]
+        %prog remove <pattern>
```

Comment:
Does the list command accept a pattern too?

Reply:
Yeah, it does. How would that be represented, since it's optional? `[<pattern>]`, `[pattern]`, or what?

Reply:
Looking at pip's style for these things I'd say `[<pattern>]` but I've no strong opinion on this.

Reply:
Switched it to `%prog list [<pattern>]`. :+1: